### ISSUE TITLE
"Automatically refresh assets in Unity" doesn't work on macOS

### ISSUE THREAD

**Issue by markusheiss** [JetBrains/resharper-unity#2361]
When I save a file in Rider, the auto refresh in Unity doesn't work automatically. I have to switch to the unity window or press the "Refresh unity assets" button in Rider to refresh the assets. The option for auto refresh is checked in Rider. I have a MacBook Pro M1. Does someone has the same issue?

Version I use:
Unity: 2021.3.11f1 Apple Silicon
Rider: 2022.2.3

<img width="797" alt="Screenshot 2022-10-09 at 10 30 26" src="https://user-images.githubusercontent.com/70845953/194746441-1e6e60ee-1a51-460f-85de-c9ffbd4bf340.png">

**Comment by van800** [comment]
How do you save a file?

**Comment by markusheiss** [comment]
I tested with ⌘-s and via menu File->Save All. nothing works

**Comment by van800** [comment]
Try to check that external Editor in Unuty is set to the correct Rider version.

**Comment by markusheiss** [comment]
external editor is set to Rider 2022.2.3 in Unity, which is my editor version

**Comment by van800** [comment]
I have just checked Refresh gets called for me with that Unity version. Unfortunately I don't have M1 at hand.
Please do the following:
1. switch the logging in Unity Preferences - Rider to Verbose, restart Unity
2. change the script in Rider and call Save All.
BTW do you see the progress like that in Rider
![image](https://user-images.githubusercontent.com/1482681/194860823-2ce31ed4-66f5-4a4c-ac49-42696beb241a.png)
Attach the log here. Thank you!

**Comment by markusheiss** [comment]
[EditorPlugin.9544.log](https://github.com/JetBrains/resharper-unity/files/9745882/EditorPlugin.9544.log)

yes, I see the progress bar. At 14:26:18.652 in the log file it says AutoRefresh is disabled via Unity settings, but in the settings it says it is enabled. Is there another Auto refresh setting?

<img width="957" alt="Screenshot 2022-10-10 at 14 29 21" src="https://user-images.githubusercontent.com/70845953/194866541-78accd1c-4da5-4605-b9ba-a823a5e0e80f.png">

**Comment by van800** [comment]
It says, that you have disable the AutoRefresh in Unity settings
```RiderPlugin                    | :1                             | AutoRefresh is disabled via Unity settings.```

**Comment by markusheiss** [comment]
my options in "General" look different to yours.

<img width="957" alt="Screenshot 2022-10-10 at 16 19 13" src="https://user-images.githubusercontent.com/70845953/194887619-0a1bd30c-8259-4cc6-9852-95938d7893a7.png">

Anyway, I got it working, but I can't really tell how. First I installed Visual Studio. It didn't work there either. Then I installed the 2020.3.40f1 Intel version of Unity. There the auto refresh was working out of the box. Then I switched back to the silicon version and it was also working there, without any changes. I then deleted the intel version and restarted the pc and it's working fine now, don't know what the problem was.
